auth/gitlab.go

```diff
@@ -0,0 +1,145 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	"fmt"
+
+	"github.com/goph/emperror"
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+	"github.com/qor/auth"
+	"github.com/spf13/cast"
+	gitlab "github.com/xanzy/go-gitlab"
+	"golang.org/x/oauth2"
+)
+
+type gitlabUserMeta struct {
+	Username  string
+	AvatarURL string
+}
+
+func NewGitlabClient(accessToken string) *gitlab.Client {
```

Comment:
Please make the GitLab API address configurable externally if it is defined in config.toml, //gittlabClient.SetBaseURL("https://git.mydomain.com/api/v3")